client/reader/README.md

```diff
@@ -9,3 +9,77 @@ These routes are served by the module:
 - /read/*
 - /recommendations (redirects to /read/search)
 - /tag/*
+
+Block Rendering Development
+======
+
+## Testing
+
+### Follow a site
+
+1. Go to `https://wordpress.com/following/manage`
+2. In the "Search or enter URL to follow..." input box, enter the website where you'll be publishing
+a block
+3. When your site appears in the search results, click the "Follow" button
+
+### Publish a post
+
+1. Go to `https://wordpress.com/home/{your_url}`
+2. Go to "Posts" on the menu to the left
+3. Click on "Add New Post"
+4. Populate your post with some content, and use the block that you're testing
+5. Click on "Publish" to publish your post or "Update" if the post already exists
+
+### Test that your block renders correctly in the Reader
+
+There's two places where your block may render.
+
+1. Go to `https://wordpress.com/home/{your_url}`
+2. Click on "Reader"
+3. Check the two tabs -- "All" and the site that you followed -- to ensure that they have rendered
+your block correctly. These are at located at the URLs `https://wordpress.com/read` and
+`https://wordpress.com/read/feeds/{site_id}` respectively. Let's call these "Reader Previews". This
+is the first place your block may render.
+4. Click into the post and ensure that the block is rendered correctly. This is located at
+`https://wordpress.com/read/feeds/{site_id}/posts/{post_id}`. Let's call these "Reader Posts". This
+is the second place your block may render.
+
+## Data Flow
+
+### Relevant files (and order of data flow)
+
+Reader Previews
+1. `client/reader/index.js`
+2. `client/reader/controller.js`
+3. `client/reader/following/main.jsx`
+4. `client/reader/stream/index.jsx`
+5. `client/reader/stream/post-lifecycle.jsx`
+6. `client/reader/stream/post.jsx`
+7. `client/blocks/reader-post-card/standard.jsx`
+8. `client/blocks/reader-excerpt/index.jsx`
+
+Reader Posts
+
+1. `client/sections.js`
+2. `client/reader/full-post/index.js`
+3. `client/reader/full-post/controller.js`
+4. `client/blocks/reader-full-post/index.jsx`
+5. `client/components/post-excerpt/index.jsx`
+
+- The last two files render the post data. It seems that the API endpoints simply return text, but
```

Comment:
Good thing is that this API response seems to include the full HTML markup ([unlike the notifications API](https://github.com/Automattic/wp-calypso/pull/45529#discussion_r486284806)):

<img width="1781" alt="Screen Shot 2020-09-11 at 11 41 38" src="https://user-images.githubusercontent.com/1233880/92905317-d999f600-f423-11ea-827b-c408e153600f.png">

This means we should be able to change the block output easily by hooking into the `render_block` filter so the block content is modified in realtime and adapted to the reader (we're exploring this same approach for emails: p1599734906020900-slack-C01ARB62VBK?thread_ts=1599683684.019300&cid=C01ARB62VBK).

Reply:
Nice! I documented what I found on the API side here https://automattictraining.wordpress.com/wp-admin/post.php?post=108464&action=edit

I guess where I'm not following is -- why would we want to modify the block content? An assumption I'm making here is that the HTML that we're receiving from the endpoint is exactly what we want and how the block is constructed. It seems like what may be missing is just the styling.

Something I also need to understand better is how we can deliver the styles that are required in the HTML that is passed to the client.